reader_concurrency_semaphore.hh

```diff
@@ -202,6 +203,15 @@ private:
     future<> enqueue_waiter(reader_permit permit, read_func func);
     void evict_readers_in_background();
     future<> do_wait_admission(reader_permit permit, read_func func = {});
+
+    // Check whether permit can be admitted or not.
+    // Caller can specify whether wait list should be empty or not for admission
+    // to be possible. can_admit::maybe means admission might be possible if some
+    // of the inactive readers are evicted.
+    enum class can_admit { yes, no, maybe };
```

Comment:
let's change the order to `{ no, maybe, yes }` so the enum integer values will make more sense.

Reply:
Done in v2.